spec/model/postgres/postgres_resource_spec.rb

```diff
@@ -24,288 +38,455 @@
+    vm_host = create_vm_host
+    storage_device = StorageDevice.create(name: "nvme0", vm_host_id: vm_host.id, total_storage_gib: 100, available_storage_gib: 80)
+    VmStorageVolume.create(vm_id: vm.id, boot: false, size_gib: 64, disk_index: 1, storage_device_id: storage_device.id)
+    server = PostgresServer.create(timeline:, resource_id: postgres_resource.id, vm_id: vm.id,
+      synchronization_status: "ready", timeline_access: "push", version: "17")
+    Strand.create_with_id(server.id, prog: "Postgres::PostgresServerNexus", label: "wait")
+
+    postgres_resource.update(ha_type: PostgresResource::HaType::NONE)
     expect(postgres_resource.has_enough_ready_servers?).to be(true)
+    postgres_resource.update(ha_type: PostgresResource::HaType::ASYNC)
     expect(postgres_resource.has_enough_ready_servers?).to be(false)
   end
 
   it "returns has_enough_ready_servers correctly when upgrading and the candidate is not present" do
-    expect(postgres_resource).to receive(:version).and_return("16")
-    expect(postgres_resource).to receive(:target_version).and_return("17")
+    # Create representative server with version 16
+    vm_rep = create_hosted_vm(project, private_subnet, "pg-vm-rep-no-candidate")
+    PostgresServer.create(timeline:, resource_id: postgres_resource.id, vm_id: vm_rep.id,
+      representative_at: Time.now, synchronization_status: "ready", timeline_access: "push", version: "16")
+    postgres_resource.update(target_version: "17")
+
+    # No upgrade_candidate_server exists (would need specific boot image setup)
     expect(postgres_resource).to receive(:upgrade_candidate_server).at_least(:once).and_return(nil)
     expect(postgres_resource.has_enough_ready_servers?).to be(false)
   end
 
   it "returns has_enough_ready_servers correctly when upgrading and the candidate is not in wait state" do
-    expect(postgres_resource).to receive(:version).and_return("16")
-    expect(postgres_resource).to receive(:target_version).and_return("17")
-    strand = instance_double(Strand, label: "wait_bootstrap_rhizome")
-    candidate_server = instance_double(PostgresServer, strand: strand, synchronization_status: "ready")
+    # Create representative server with version 16
+    vm_rep = create_hosted_vm(project, private_subnet, "pg-vm-rep-not-ready")
+    PostgresServer.create(timeline:, resource_id: postgres_resource.id, vm_id: vm_rep.id,
+      representative_at: Time.now, synchronization_status: "ready", timeline_access: "push", version: "16")
+    postgres_resource.update(target_version: "17")
+
+    vm = create_hosted_vm(project, private_subnet, "pg-vm-candidate-not-ready")
+    candidate_server = PostgresServer.create(timeline:, resource_id: postgres_resource.id, vm_id: vm.id,
+      synchronization_status: "ready", timeline_access: "push", version: "17")
+    Strand.create_with_id(candidate_server.id, prog: "Postgres::PostgresServerNexus", label: "wait_bootstrap_rhizome")
```

Comment:
```suggestion
    Strand.create_with_id(candidate_server, prog: "Postgres::PostgresServerNexus", label: "wait_bootstrap_rhizome")
```

Similar issue elsewhere.